api/python/provisioner/commands/check.py

```diff
@@ -0,0 +1,578 @@
+                    log_filename=log_file,
+                    key_phrase=phrase)
+                try:
+                    cmd_run(cmd, targets=_targets)
+                except SaltCmdResultError as e:
+                    _res.set_fail(target=minion_id,
+                                  comment=f"{log_file}: {str(e.reason)}")
+                    return _res
+
+            _res.set_passed(target=minion_id, comment=f"{log_file}")
+
+            return _res
+
+        # grep "unable to stop resource" /var/log/pacemaker.log
+        # grep "reboot" /var/log/corosync.log
+        # grep "error" /var/log/corosync.log"
+
+        # they should be a constants
+        corosync_log = "/var/log/cluster/corosync.log"
+        pacemaker_log = "/var/log/pacemaker.log"
+
+        corosync_key_phrases = (
+            "reboot",
+            "error"
+        )
+        pacemaker_key_phrases = (
+            "unable to stop resource",
+        )
+
+        res.append(
+            check_log_file(corosync_log, corosync_key_phrases, minion_id))
+        res.append(
+            check_log_file(pacemaker_log, pacemaker_key_phrases, minion_id))
+
+        return res
+
+    @staticmethod
+    def _passwordless_ssh_access(
+                            *,
+                            servers: Union[list, tuple, set] = PENDING_SERVERS,
+                            args: str) -> CheckEntry:
+        """
+        Check if passwordless SSH access is possible between nodes
+
+        :param args: passwordless ssh access check specific parameters
+                     and arguments
+        :param servers: servers for checking passwordless SSH
+        :return:
+        """
+        # NOTE: the logic is similar to `_connectivity` method
+        _PSWDLESS_SSH_CHECK_CMD_TMPL = (
+                                    'ssh -o BatchMode=yes '
+                                    '-o PasswordAuthentication=no '
+                                    '{user}@{hostname} exit 0 &>/dev/null; '
+                                    'test $? == 0 || {{ echo "fail" && '
+                                    'exit 1; }}')
+
+        res: CheckEntry = CheckEntry(cfg.Checks.PASSWORDLESS_SSH_ACCESS.value)
+
+        user = "root"
+        for addr in servers:
+            # NOTE: check ping of 'srvnode-2' from 'srvnode-1'
+            # and vise versa
+            # TODO: which targets do we need to use? Because we need to
+            #  check cross connectivity if nodes > 2
+            targets = set(servers) - {addr}
+
+            targets = (cfg.LOCAL_MINION if not targets
+                       else next(iter(targets)))  # takes just one node
+
+            cmd = _PSWDLESS_SSH_CHECK_CMD_TMPL.format(user=user,
+                                                      hostname=addr)
+            try:
+                _res = cmd_run(cmd, targets=targets)
+            except SaltCmdResultError:
+                res.set_fail(target=targets,
```

Comment:
the same concern as above for ping command

Reply:
Done